command/sub8_missions/missions/buoy.py

```diff
@@ -1,89 +1,55 @@
-    if response.found:
-        align(sub)
-        yield sub.move.forward(0.4).go(speed=0.1)
-    util.defer.returnValue(response.found)
-
-
-@util.cancellableInlineCallbacks
-def bump(sub):
-    response = yield sub.buoy.get_pose()
-    transform = yield sub._tf_listener.get_transform(
-        '/map',
-        '/stereo_front',
-        response.pose.header.stamp)
-    yield sub.move.height(response.pose.z).go()
-
-
-@util.cancellableInlineCallbacks
-def get_buoy_position(sub):
-    transform = yield sub._tf_listener.get_transform(
-        '/map',
-        '/stereo_front',
-        response.pose.header.stamp
-    )
-    tft = tf.Transform.from_Pose_message(response.pose.pose)
-    full_transform = transform * tft
-    util.defer.returnValue(full_transform)
-
-
-@util.cancellableInlineCallbacks
 def run(sub):
     # buoy_search is a Deferred
     print "We're looking for a buoy"
+
+    print "Executing search pattern"
+    yield sub.move.right(2.0).go()
+    yield sub.move.down(0.3).go()
+
+    print "Now trying to pose it"
     response = yield sub.buoy.get_pose('red')
     if not response.found:
         print 'failed to discover buoy location'
         return
+    else:
+        print "Got buoy pose"
+        print response.pose
 
     back = sub.move.forward(0.1)
-    transform = yield sub._tf_listener.get_transform(
-        '/map',
-        '/stereo_front',
-        response.pose.header.stamp
-    )
-    tft = tf.Transform.from_Pose_message(response.pose.pose)
-    full_transform = transform * tft
-    print full_transform._p
-    # yield sub.move.set_position(full_transform._p).go()
+
+    full_transform = tf.Transform.from_Pose_message(response.pose.pose)
     print 'setting height'
     if full_transform._p[2] > -0.2:
         print 'Detected buoy above the water'
-        return
+        defer.returnValue(False)
 
-    yield sub.move.height(full_transform._p[2]).go(speed=0.2)
+    yield sub.move.depth(-full_transform._p[2]).go(speed=0.2)
     yield util.wall_sleep(0.2)
     print 'looking at'
-    yield sub.move.look_at(full_transform._p).go(speed=0.1)
-    print 'position'
+    yield sub.move.look_at_without_pitching(full_transform._p).go()
     yield util.wall_sleep(0.2)
-    # go = clip_norm(transform._p, 0.01, )
-    dist = response.pose.pose.position.z
-    print 'd', dist
-    if dist > 1.0:
-        print 'moving forward'
-        yield sub.move.forward(dist - 1.0).go(speed=0.2)
 
-    # yield sub.move.forward(0.2).go(speed=0.1)
+    yield sub.move.look_at_without_pitching(full_transform._p).go()
+    yield util.wall_sleep(0.2)
+
+    yield sub.move.look_at_without_pitching(full_transform._p).go()
+    yield util.wall_sleep(0.2)
+
+    dist = np.inf
+    while(dist > 1.0):
+        dist = np.linalg.norm(full_transform._p - sub.pose.position) * 0.5
+        yield sub.move.look_at_without_pitching(full_transform._p).go()
+        yield util.wall_sleep(0.2)
+
+        yield sub.move.forward(dist).go()
```

Comment:
Why not just move right up next to it the estimated position in one fell swoop here? 
It seems like multiple iterations would only help if we re-estimated the position of the bouy each time we moved.